docs/articles/documentation/continuous-integration/jenkins.md

```diff
@@ -11,9 +11,108 @@ TestCafe has an extensive command line interface that allows it to fit well in a
+
+> Important! The Jenkins installer overwrites permission settings during upgrade. Download and replace `jenkins.war` to upgrade Jenkins, or repeat the steps below after you run the installer.
+
+### Obtain Permissions on macOS
+
+1. Stop Jenkins:
+
+    ```sh
+    launchctl unload /Library/LaunchDaemons/org.jenkins-ci.plist
+    ```
+
+2. Open the `org.jenkins-ci.plist` file in a text editor:
+
+    ```sh
+    sudo nano /Library/LaunchDaemons/org.jenkins-ci.plist
+    ```
+
+3. Find the `USER_NAME` option and change its value to the user name under whose account macOS is started:
+
+    ```xml
+    <key>USER_NAME</key>
+    <string>peter.p</string>
+    ```
+
+4. Find the `JENKINS_HOME` key that specifies the Jenkins home directory. Copy the directory content to a local user's home directory and specify the new path under the `JENKINS_HOME` key:
+
+    ```xml
+    <key>JENKINS_HOME</key>
+    <string>/users/peter.p/jenkins/</string>
+    ```
+
+    Close the file and save your changes.
+
+5. Change the Jenkins temporary directory:
+
+    ```sh
+    sudo defaults write /Library/Preferences/org.jenkins-ci tmpdir "/tmp"
+    ```
+
+6. Change the `/var/log/jenkins` file ownership to your account:
+
+    ```sh
+    sudo chown peter.p /var/log/jenkins
+    ```
+
+7. Start Jenkins:
+
+    ```sh
+    launchctl load /Library/LaunchDaemons/org.jenkins-ci.plist
+    ```
+
+### Obtain Permissions on Linux
+
+1. Stop Jenkins:
+
+    ```sh
+    sudo systemctl stop jenkins.service
+    ```
+
+2. Open the `/etc/default/jenkins` file in a text editor:
+
+    ```sh
+    sudo nano /etc/default/jenkins
+    ```
+
+3. Specify your user name in the `JENKINS_USER` variable:
+
+    ```text
+    JENKINS_USER=peter.p
+    ```
+
+    Then close the file and save your changes.
+
+4. Change the Jenkins directory ownership to your account:
+
+    ```sh
+    sudo chown peter.p /var/log/jenkins
+    sudo chown peter.p /var/lib/jenkins
+    sudo chown peter.p /var/run/jenkins
+    sudo chown peter.p /var/cache/jenkins
+    ```
+
+5. Start Jenkins:
+
+    ```sh
+    sudo systemctl start jenkins.service
+    ```
+
+## Step 1 - Install Node.js Plugin
+
+1. Follow the [instructions from Jenkins documentation](https://jenkins.io/doc/book/managing/plugins/#installing-a-plugin) to install a [Node.js plugin](https://plugins.jenkins.io/nodejs).
+
+2. Configure this plugin as described in the [Usage section](https://plugins.jenkins.io/nodejs#NodeJSPlugin-Usage) on the plugin page. Ensure that you have added a Node.js installation on the **Global Tool Configuration** page and checked the **Provide Node & npm bin/ folder to PATH** check box in your build configuration.
+
+## Step 2 - Fetch Test Code From a Repository
+
+This tutorial uses tests published in the [ci-integration-demo](https://github.com/VasilyStrelyaev/ci-integration-demo) GitHub repository. If you use a different version control system, search for a plugin that integrates it with Jenkins.
```

Comment:
`search for a plugin` search where? I'd get rid of this sentence and/or rephrase like 
"E.g. for GitHub repository... [ci-integration-demo]"
Also shouldn't we move the demo to devexpress repo, phrobably we need a new repository for examples?

Reply:
Filed https://github.com/DevExpress/testcafe/issues/3726 to move the demo.

Reply:
good